Gruntfile.js

```diff
@@ -1,13 +1,86 @@
 module.exports = function(grunt) {
-    // Project configuration
+
     'use strict';
-    grunt.loadNpmTasks('grunt-contrib-requirejs');
-    grunt.loadNpmTasks('grunt-filerev');
-    grunt.loadNpmTasks('grunt-regex-replace');
-    grunt.loadNpmTasks('grunt-karma');
-    grunt.loadNpmTasks('grunt-coveralls');
-    grunt.loadNpmTasks('grunt-jasmine-nodejs');
-    grunt.loadNpmTasks('grunt-contrib-uglify-es');
+
+    require('load-grunt-tasks')(grunt);
+
+    // files required for the 'page.html' template
+    const kbaseStaticDir = 'kbase-extension/static/kbase/',
+
+    staticFilesNoConcat = [
+        kbaseStaticDir + 'custom/*.css',
+        kbaseStaticDir + 'css/*.css',
+        '!' + kbaseStaticDir + 'css/*_concat.css',
+    ],
+
+    pageCssFiles = [
+        "custom/custom.css",
+        "css/kbaseStylesheet.css",
+        "css/kbaseNarrative.css",
+        "css/kbaseIcons.css",
+        "css/landingPages.css",
+        "css/kbaseEditor.css",
+        "css/kbaseNotify.css",
+        "css/methodCell.css",
+        "css/bootstrapHelper.css",
+        "css/buttons.css",
+        "css/contigBrowserStyles.css", // may need to move this out
+        "css/kbaseJobLog.css",
+        "css/kbaseTour.css",
+    ].map( s => kbaseStaticDir + s ),
+
+    dynamicallyLoadedCssFiles =         [
+        "css/appCell.css",
+        "css/advancedViewCell.css",
+        "css/editorCell.css",
+    ].map( s => kbaseStaticDir + s ),
+
+    allCssFiles = pageCssFiles.concat(dynamicallyLoadedCssFiles),
+
+    cssFileHeader = '/** This file is generated automatically; edits will not be saved.\n'
+    + 'Please edit the original source files in kbase-extension/static/kbase/* '
+    + 'and run "grunt concat" to regenerate this file. */\n\n',
+
+    // stylelint options
+    stylelintOptions = {
+        configFile: '.stylelintrc.json',
+        formatter: 'string',
+        ignoreDisables: false,
+        failOnError: false,
+        reportNeedlessDisables: true,
+        fix: true,
+    },
+
+    // clean up existing css - edits files in place
+    cleanExistingCssOptions = {
+        processors: [
+            // remove vendor prefixes
+            require('postcss-unprefix')(),
+            require('postcss-remove-prefixes')(),
+            // convert string colour names to hex codes for consistency
+            require('postcss-color-converter')({
+                outputColorFormat: 'hex',
+                ignore: ['rgb'],
```

Comment:
Why is rgb being ignored?

Reply:
Somewhat arbitrary -- the main thing I wanted to do was to convert text colour names to hex codes. The minifier [https://cssnano.co](cssnano) has a setting that converts colours into whatever the shortest representation is -- it might be best to use that.

Reply:
I just found out that it throws an error when trying to convert RGB colours, so that was a lucky choice!

Reply:
On second thoughts, I'm going to remove this and use a stylelint plugin to convert everything to RGB, since RGB also allows transparency.